test, [: add page, improve examples (#5205)
tldr-pages/tldr: pages/common/[.md

> # test

Reviewer: Oops, yeah @bl-ue! I've just noticed this. We should update this.
```suggestion
# [
```

Reviewer 2: Issue is though, the linter doesn't like that, see below?

> Evaluate condition.

> > If it is true, returns 0 exit status, otherwise returns 1.

Reviewer 2: ```suggestion
> Returns 0 if the condition evaluates to true, 1 if it evaluates to false.
```

> More information: <https://www.gnu.org/software/coreutils/manual/html_node/test-invocation.html>.

> - Test if given variable is equal to given string:

Reviewer: Perhaps this will read better?
```suggestion
- Test if a given variable is equal to a given string:
```

> `[ $MY_VAR == '/bin/zsh' ]`

Reviewer 2: ```suggestion
`[ "{{$MY_VAR}}" == "{{/bin/zsh}}" ]`
```

> - Test if given variable is empty:

Reviewer: Perhaps this will read better?
```suggestion
- Test if a given variable is empty:
```

> `[ -z $GIT_BRANCH ]`

Reviewer 2: ```suggestion
`[ -z "{{$GIT_BRANCH}}" ]`
```

> - Test if file exists:

Reviewer: Perhaps this will read better?
```suggestion
- Test if a file exists:
```

> `[ -e {{filename}} ]`

Reviewer 2: ```suggestion
`[ -f "{{path/to/file_or_directory}}" ]`
```

> - Test if directory not exists:

Reviewer: Perhaps this will read better?
```suggestion
- Test if a directory does not exist:
```

> `[ ! -d {{path/to/directory}} ]`

Reviewer 2: ```suggestion
`[ ! -d "{{path/to/directory}}" ]`
```

- If-else statement:

> `[ {{condition}} ] && echo "true" || echo "false"`

Reviewer 2: ```suggestion
`[ {{condition}} ] && {{echo "true"}} || {{echo "false"}}`
```
